pyccl/tests/test_power.py

```diff
@@ -32,7 +32,7 @@ def test_halomod_f2d_copy():
         ccl.CCLWarning,
         ccl.nonlin_matter_power, COSMO_HM, 1., 0.8)
     pk_new = pk2d.eval(1., 0.8, COSMO_HM)
-    psp_old = COSMO_HM.cosmo.data.p_nl
+    psp_old = COSMO_HM._pk_nl['delta_matter_x_delta_matter'].psp
```

Comment:
Private attribute access is a bad sign. I think the `get_nonlin_power` method I suggested above might be a good idea.

Reply:
Fixed